rootex/core/renderer/material.h

```diff
@@ -21,6 +23,13 @@ class Material
 	Material();
 	virtual ~Material() = default;
 	virtual void bind();
+	// TODO- Make this purely virtual
+	virtual void draw() = 0;
+	String getFileName() { return m_FileName; };
+	String getTypeName() { return m_TypeName; };
+	String getFullName() { return m_FileName + " - " + m_TypeName; };
+	void setFileName(String& fileName) { m_FileName = fileName; };
+	virtual JSON::json getJSON() const = 0;
```

Comment:
Define getJSON() here that serializes m_FileName and m_Type. Now in ColorMaterial::getJSON() and other children of Material, do
```cpp
JSON::json ColorMaterial getJSON()
{
    JSON::json& j = Material::getJSON();
    // Add some things to `j` that only belong to ColorMaterial and not the parent class.
    return j;
}
```